services/src/main/java/org/keycloak/protocol/ciba/EarlyAccessBlockerStore.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.protocol.ciba;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.keycloak.common.util.Time;
+import org.keycloak.protocol.ciba.utils.EarlyAccessBlocker;
+
+public class EarlyAccessBlockerStore {
+
+    private final static ConcurrentMap<UUID, EarlyAccessBlockerValueEntity> blockerCache = new ConcurrentHashMap<>();
```

Comment:
Maybe a `LinkedHashMap` makes sense here so we could bound the map to size and avoid memory leaks.

It should also be possible to remove the eldest entry more easily. Instead of iterating over entries (mainly if it is expected a potential growth in entries).

Reply:
@pedroigor Thank you. I would like to try to use `LinkedHashMap`.